GliaWidgets/Sources/EntryWidget/EntryWidget.swift

```diff
@@ -0,0 +1,162 @@
+import Foundation
+import UIKit
+import SwiftUI
+
+public final class EntryWidget: NSObject {
+    private var hostedViewController: UIViewController?
+    private var embeddedView: UIView?
+    private var queueIds: [String] = []
+
+    // Channels will become dynamic in the subsequent PRs
+    private var channels: [Channel] = [.chat, .audio, .video, .secureMessaging]
+    private let theme: Theme
+    private let sizeConstraints: SizeConstraints = .init(
+        singleCellHeight: 72,
+        singleCellIconSize: 24,
+        poweredByContainerHeight: 40,
+        sheetHeaderHeight: 36,
+        sheetHeaderDraggerWidth: 32,
+        sheetHeaderDraggerHeight: 4,
+        dividerHeight: 1
+    )
+
+    init(theme: Theme) {
+        self.theme = theme
+    }
+
+    public func show(by presentation: EntryWidget.Presentation) {
+        switch presentation {
+        case let .sheet(parentViewController):
+            showSheet(in: parentViewController)
+        case let .embedded(parentView):
+            showView(in: parentView)
+        }
+    }
+
+    func hide() {
+        hostedViewController?.dismiss(animated: true, completion: nil)
+        hostedViewController = nil
+        embeddedView?.removeFromSuperview()
+        embeddedView = nil
+    }
+
+    func createEntryWidgetView(
+        queueIds: [String],
+        channels: [Channel] = [.chat, .audio, .video]
+    ) {
+        self.queueIds = queueIds
+        self.channels = channels
+    }
+}
+
+private extension EntryWidget {
+    func showView(in parentView: UIView) {
+        parentView.subviews.forEach { $0.removeFromSuperview() }
+        let model = makeViewModel(
+            channels: channels,
+            selection: { channel in
+                // Logic for handling this callback will be handled later
```

Comment:
Ideally for such cases it would be beneficial to add ticket number to avoid missing this later.